ocl/mappings/models.py

```diff
@@ -251,8 +248,12 @@ def persist_new(cls, mapping, created_by, **kwargs):
         mapping.parent = parent_resource
         mapping.public_access = parent_resource.public_access
 
+        # Perform mapping validations
+        from validators import MappingValidator
         try:
             mapping.full_clean()
```

Comment:
full_clean is expected to do any validation on the object. It shouldn't be required to call MappingValidator explicitly...

Reply:
The reason I excluded this validation from full_clean is to distinguish it from create and update method. I wanted to restrict the creation of same mapping within parent but not to restrict it for update method. Since both update method also calls full_clean, this creates obstacle on updating the object because of the validation restriction.

Reply:
Just add a filter on id in the validation query to skip the current object if it was saved.